### ISSUE TITLE
Transaction Costs Optimization

### ISSUE THREAD

**Issue by DrPaprikaa** [Marigold/universal-portfolios#45]
Hi,

Correct me if I'm wrong, but the various algos implemented in this wonderful repo have been tested in the litterature in the framework of non-existent transaction costs.

[This paper](https://ink.library.smu.edu.sg/cgi/viewcontent.cgi?referer=&httpsredir=1&article=4761&context=sis_research) improves existing OLPS strategies in the case of non-zero transaction costs, maybe you've already read it. Using their words : 

> We add a second term that minimizes the L1 norm of difference between two consecutive allocations,which is equivalent to minimizing the incurred transaction costs.

Their implementation seems effective, what do you think about it ? Could it be implemented here ?

I tried an implementation in OLMAR's `update()` for example. It goes along like this :
        
```python

def update(b, x_pred):
    window = 12
    n = 10
    lam = 10*trx_fee_pct

    # Calculate variables
    vt   = x_pred / np.dot(b, x_pred)
    v_t_ = np.dot(1, vt) / window

    # Update portfolio
    b_1 = n * (vt - np.dot(v_t_, 1))
    b_  = b_1 + np.sign(b_1)*np.maximum(np.zeros(len(b_1)), np.abs(b_1) - lam)

    # project it onto simplex
    return tools.simplex_proj(b_)
```

Thanks !
Let me know if can help more :)

**Comment by Marigold** [comment]
> Their implementation seems effective, what do you think about it ? Could it be implemented here ?

Yeah, adding a parameter for controlling transaction costs would be cool. Could you open a PR for that? I'll be happy to review it.

**Comment by DrPaprikaa** [comment]
I opened a PR with an implementation of my interpretation of TCO, which needs checking. Cheers :)

**Comment by DrPaprikaa** [comment]
Closing this issue, TCO has been implemented in [#47](https://github.com/Marigold/universal-portfolios/pull/47).
